core/src/main/scala/org/apache/spark/security/SocketAuthServer.scala

```diff
@@ -66,6 +66,21 @@ private[spark] abstract class SocketAuthServer[T](
 
 }
 
+/**
+ * Create a socket server class and run user function on the socket in a background thread.
+ * This is the same as calling SocketAuthServer.setupOneConnectionServer except it creates
+ * a server object that can then be synced from Python.
+ */
+private [spark] class SocketFuncServer(
+    authHelper: SocketAuthHelper,
+    threadName: String,
+    func: Socket => Unit) extends SocketAuthServer[Unit](authHelper, threadName) {
```

Comment:
I don't think we need `SockAuthServer.setupOneConnectionServer` if we have this also, so it could be cleaned up

Reply:
I removed `SockAuthServer.setupOneConnectionServer` and replaced usage with SocketFuncServer